src/compiler/checker.ts

```diff
@@ -2913,6 +2915,18 @@ namespace ts {
             }
         }
 
+        function areAllOuterTypeParametersApplied(type: Type): boolean {
+            const outerTypeParameters = (<InterfaceType>type).outerTypeParameters;
+            if (outerTypeParameters) {
+                // an unapplied type parameter is one 
+                // whose argument symbol is still the same as the parameter symbol
+                const last = outerTypeParameters.length - 1;
+                const typeArguments = (<TypeReference>type).typeArguments;
+                return outerTypeParameters[last].symbol !== typeArguments[last].symbol;

```

Comment:
Can you leave a comment about why we can get away with just checking the last type parameter? It's not necessarily clear to someone stumbling upon this code for the first time.